core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -29,7 +30,10 @@ import org.apache.spark._
 import org.apache.spark.internal.Logging
 import org.apache.spark.util.{RedirectThread, Utils}
 
-private[spark] class PythonWorkerFactory(pythonExec: String, envVars: Map[String, String])
+
+private[spark] class PythonWorkerFactory(var pythonExec: String,
```

Comment:
Making pythonExec a var is maybe not ideal.

Reply:
Fixed